clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -214,11 +215,50 @@ public String toString() {
         private final boolean shallow;
         private RecordsIterator innerIter;
 
+        // The variables for inner iterator
+        private final LinkedList<LogEntry> logEntries;
```

Comment:
Why we are using a LinkedList here? It rarely makes sense to use it, even the person who wrote it says so (https://twitter.com/joshbloch/status/583813919019573248).